go-controller/pkg/node/gateway_udn.go

```diff
@@ -717,3 +798,84 @@ func addRPFilterLooseModeForManagementPort(mgmtPortName string) error {
 	}
 	return nil
 }
+
+func (udng *UserDefinedNetworkGateway) run() {
+	go func() {
+		for range udng.reconcile {
+			err := retry.OnError(
+				wait.Backoff{
+					Duration: 10 * time.Millisecond,
+					Steps:    4,
+					Factor:   5.0,
+				},
+				func(error) bool {
+					select {
+					case _, open := <-udng.reconcile:
+						return open
+					default:
+						return true
+					}
+				},
+				udng.doReconcile,
+			)
+			if err != nil {
+				klog.Errorf("Failed to reconcile gateway for network %s: %v", udng.GetNetworkName(), err)
+			}
+		}
+	}()
+}
+
+func (udng *UserDefinedNetworkGateway) Reconcile() {
+	select {
+	case udng.reconcile <- struct{}{}:
+	default:
+	}
+}
+
+func (udng *UserDefinedNetworkGateway) doReconcile() error {
```

Comment:
how come this reconcile was implemented asynchronously? It looks like the calls inside of it are not expensive or blocked for long periods of time. The side effect of this being async is on node start up, the syncall method for networks will not be blocked on the reconcile.

Reply:
This is not used on startup. Everything that needs to happen on startup happens synchronously in `AddNetwork`.

Reply:
It would be

func (c *networkController) Start() -> c.syncAll -> err := c.syncNetwork(network.GetNetworkName()) -> err = err = c.ensureNetwork(want) -> err := networkController.Start() -> oc.gateway.AddNetwork()

Reply:
Yeah, I think we should. In general terms Start should just get things started and have minimum surface for failure. However that requires not having dependencies to do things synchronously. Too much for this PR.